### DISCUSSION TITLE
task manager could not pick up icon for gnumeric

### DISCUSSION THREAD

**Discussion by andy-chung** [lxqt/lxqt#2529] [Q&A]
##### Steps to Reproduce (for bugs)

1. config task manager. check "use icons by windows class, if avail"
2. verify that gnumeric icon avail at /usr/share/icons//hicolor/16x16/apps/org.gnumeric.gnumeric.png ... 64x64
3. verify that .local/share/applications/gnumeric desktop file has "StartupWMClass=org.gnumeric.gnumeric"
4. launch gnumeric
5. task bar does not show gnumeric icon but the default rectangle with blue bar on top

##### System Information

* Distribution & Version: fedora 39 lxqt
* Kernel: 6.7.4-200.fc39.x86_64
* Qt Version: 
* liblxqt Version: 1.3.0
* Package version: 

![Screenshot from 2024-02-18 18-25-40](https://github.com/lxqt/lxqt/assets/5138820/fad2d05b-8735-4c5d-b106-e3fe40634858)

**Comment by tsujan** [comment]
What happens is what you see: you have no icon with the name of Gnumeric.'s window class. Find what it is and add an icon with that name.

**Comment by andy-chung** [comment]
Thanks. not sure if i get it right....

1. "org.gnumeric.gnumeric" got from xprop WM_CLASS

```
andy@MiWiFi-RM1800-srv:~$ xprop WM_CLASS
WM_CLASS(STRING) = "org.gnumeric.gnumeric", "Gnumeric"

```
2. icon files are here
```
andy@MiWiFi-RM1800-srv:/usr/share/icons$ find . -name "org.gnum*.*"
./hicolor/16x16/apps/org.gnumeric.gnumeric.png
./hicolor/22x22/apps/org.gnumeric.gnumeric.png
./hicolor/24x24/apps/org.gnumeric.gnumeric.png
./hicolor/256x256/apps/org.gnumeric.gnumeric.png
./hicolor/32x32/apps/org.gnumeric.gnumeric.png
./hicolor/48x48/apps/org.gnumeric.gnumeric.png
./hicolor/64x64/apps/org.gnumeric.gnumeric.png
```


3. desktop file as below
```
-Desktop-File-Install-Version=0.26
StartupWMClass=org.gnumeric.gnumeric

```

**Comment by tsujan** [comment]
It's probably `gnumeric.png`.

LXQt Panel uses KF5 to get it: https://github.com/lxqt/lxqt-panel/blob/fd2fb3cb15b0ab996ebc5ac6139d89739ca3dfb2/plugin-taskbar/lxqttaskbutton.cpp#L144-L157

**Comment by andy-chung** [comment]
tried duplicate a set of gnumeric.org ... but yet without success ... icon not shown in task bar ...

```
andy@MiWiFi-RM1800-srv:/usr/share/icons/hicolor$ find . -name "*gnumer*"
./16x16/apps/org.gnumeric.gnumeric.png
./16x16/apps/gnumeric.png
./22x22/apps/org.gnumeric.gnumeric.png
./22x22/apps/gnumeric.png
./24x24/apps/org.gnumeric.gnumeric.png
./24x24/apps/gnumeric.png
./256x256/apps/org.gnumeric.gnumeric.png
./256x256/apps/gnumeric.png
./32x32/apps/org.gnumeric.gnumeric.png
./32x32/apps/gnumeric.png
./48x48/apps/org.gnumeric.gnumeric.png
./48x48/apps/gnumeric.png
./64x64/apps/org.gnumeric.gnumeric.png
./64x64/apps/gnumeric.png
```

**Comment by tsujan** [comment]
Did you update the icon cache or restart the panel after that?

**Comment by andy-chung** [comment]
oops .... logout and login ... now it works ... thanks.

**Reply by tsujan** [reply]
> logout and login ... now it works

Good to know that.

To restart LXQt components you could just use LXQt Session Settings; no need to logging out and in.
